data_structures/binary_tree/binary_search_tree.py

```diff
@@ -20,6 +77,10 @@ def __repr__(self) -> str:
             return str(self.value)
         return pformat({f"{self.value}": (self.left, self.right)}, indent=1)
 
+    @property
+    def is_right(self):
+        return self.parent and self is self.parent.right
```

Comment:
I think this should fix the return type error while keeping the return type annotation
```suggestion
    def is_right(self) -> bool:
        return self.parent is not None and self is self.parent.right
```